lib/src/facts/windows/dmi_resolver.cc

```diff
@@ -18,11 +18,19 @@ namespace facter { namespace facts { namespace windows {
         data result;
 
         auto vals = _wmi->query(wmi::computersystemproduct, {wmi::name});
-        result.product_name = wmi::get(vals, wmi::name);
+        if (vals.empty()) {
+            LOG_DEBUG("failure resolving dmi::product::name fact, wmi could not resolve %1%::%2%", wmi::computersystemproduct, wmi::name);
+        } else {
+            result.product_name = wmi::get(vals, wmi::name);
+        }
 
         vals = _wmi->query(wmi::bios, {wmi::manufacturer, wmi::serialnumber});
-        result.serial_number = wmi::get(vals, wmi::serialnumber);
-        result.manufacturer = wmi::get(vals, wmi::manufacturer);
+        if (vals.empty()) {
+            LOG_DEBUG("failure resolving dmi::{serialnumber,manufacturer} facts, wmi could not resolve %1%", wmi::manufacturer);
```

Comment:
It may not be worth including the details about wmi, as they'd be obvious to us. It might be useful to Windows devs to know that wmi isn't behaving correctly. Have a suggestion on improving it?